File: app/src/main/java/com/frequencias/formas/MainActivity.java
package com.frequencias.formas;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.ViewGroup;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private WebView webView;
    private static final String FREQ_COMPLETO_URL = "file:///android_asset/freq_completo.html";
    private static final String SHOTS_EMOCIONAIS_URL = "file:///android_asset/shots_emocionais.html";
    // A tela principal precisa ser a experiência com áudio e visualização animada.
    private String currentUrl = FREQ_COMPLETO_URL;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        webView = new WebView(this);
        webView.setLayoutParams(new ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT));
        setContentView(webView);

        WebSettings settings = webView.getSettings();
        settings.setJavaScriptEnabled(true);
        settings.setDomStorageEnabled(true);
        settings.setDatabaseEnabled(true);
        settings.setMediaPlaybackRequiresUserGesture(false);
        settings.setAllowFileAccess(true);
        settings.setAllowContentAccess(true);
        settings.setCacheMode(WebSettings.LOAD_DEFAULT);
        settings.setUseWideViewPort(true);
        settings.setBuiltInZoomControls(true);

        webView.setWebViewClient(new WebViewClient());
        webView.setWebChromeClient(new WebChromeClient());

        // A página principal contém o player e a animação em canvas.
        webView.loadUrl(currentUrl);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        menu.add(Menu.NONE, 1, Menu.NONE, "📊 Shots Emocionais")
                .setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS);
        menu.add(Menu.NONE, 2, Menu.NONE, "🎵 Frequências")
                .setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == 1) {
            currentUrl = SHOTS_EMOCIONAIS_URL;
            webView.loadUrl(currentUrl);
            return true;
        } else if (item.getItemId() == 2) {
            currentUrl = FREQ_COMPLETO_URL;
            webView.loadUrl(currentUrl);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onBackPressed() {
        if (webView.canGoBack()) {
            webView.goBack();
        } else {
            super.onBackPressed();
        }
    }
}

File: app/src/main/kotlin/com/binaural/shots/MainActivity.kt
package com.binaural.shots

import android.os.Bundle
import android.webkit.WebSettings
import android.webkit.WebView
import androidx.appcompat.app.AppCompatActivity
import android.webkit.WebViewClient

class MainActivity : AppCompatActivity() {

    private lateinit var webView: WebView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        webView = findViewById(R.id.webView)
        
        // Configurar o WebView
        with(webView.settings) {
            javaScriptEnabled = true
            domStorageEnabled = true
            databaseEnabled = true
            cacheMode = WebSettings.LOAD_DEFAULT
            
            // Para melhor performance
            setAppCacheMaxSize(10 * 1024 * 1024) // 10MB
            setAppCachePath(cacheDir.path)
            setAppCacheEnabled(true)
            
            // Permitir acesso a mídia
            mediaPlaybackRequiresUserGesture = false
            
            // Layout
            useWideViewPort = true
            loadWithOverviewMode = true
            
            // Zoom
            builtInZoomControls = true
            displayZoomControls = false
            
            // User agent
            userAgentString = userAgentString + " BinauralShots/1.0"
        }

        // Definir client para controlar navegação
        webView.webViewClient = WebViewClient()

        // Carregar a página HTML inicial (mapa emocional)
        webView.loadUrl("file:///android_asset/shots_emocionais.html")
    }

    override fun onBackPressed() {
        if (webView.canGoBack()) {
            webView.goBack()
        } else {
            super.onBackPressed()
        }
    }
}
